Sources/Preferences/SwiftUIComponents.swift

```diff
@@ -0,0 +1,128 @@
+//
+//  SwiftUIComponents.swift
+//  Preferences
+//
+//  Created by Kacper on 09/01/2020.
+//
+
+import SwiftUI
+
+/**
+    Extension with custom alignment guide for section title labels.
+ */
+@available (macOS 10.15, *)
+extension HorizontalAlignment {
+    private enum PreferenceSectionLabelAlignment: AlignmentID {
+        static func defaultValue(in context: ViewDimensions) -> CGFloat {
+            return context[HorizontalAlignment.leading]
+        }
+    }
+    
+    static let preferenceSectionLabel = HorizontalAlignment(PreferenceSectionLabelAlignment.self)
+}
+
+/**
+    Function builder for Preferences component used in order to restrict types of child view to PreferenceSection.
+*/
+@available (macOS 10.15, *)
+@_functionBuilder
+public struct SectionBuilder {
+    public static func buildBlock(_ sections: PreferenceSection...) -> [PreferenceSection] {
+        return sections
+    }
+}
+
+/**
+	Represents section with left aligned title and optional bottom divider (present by default).
+ */
+@available (macOS 10.15, *)
+public struct PreferenceSection: View   {
+    public let title: String
+    public let content: () -> AnyView
+    public let bottomDivider: Bool
+    
+    public init<Content: View>(title: String, bottomDivider: Bool = true, content: @escaping () -> Content) {
+        self.title = title
+        self.bottomDivider = bottomDivider
+        self.content = { AnyView(content()) }
```

Comment:
Do we have to use `AnyView` here?

Reply:
In order for PreferenceSectionBuilder to work and output array of sections, I had to type-erase the content.